lm_eval/models/huggingface.py

```diff
@@ -727,6 +727,11 @@ def _model_generate(self, context, max_length, stop, **generation_kwargs):
         # and we don't want a warning from HF
         generation_kwargs["temperature"] = generation_kwargs.get("temperature", 0.0)
         do_sample = generation_kwargs.get("do_sample", None)
+
+        # The temperature has to be a strictly positive float -- if it is 0.0, use greedy decoding strategies
+        if generation_kwargs.get("temperature") == 0.0 and do_sample is None:
+            do_sample = False
```

Comment:
is `do_sample` ‘s value getting passed to generate here? maybe I’m just misreading but it never gets added back into generation_kwargs here?

Reply:
Sorted! I'm running some tests locally to check if this completely solves the problem, will get back shortly